web/src/ee/features/billing/server/cloudBillingRouter.ts

```diff
@@ -233,102 +377,682 @@ export const cloudBillingRouter = createTRPCRouter({
             subscription.status,
         });
 
-      if (subscription.items.data.length !== 1)
+      const newProduct = await client.products.retrieve(input.stripeProductId);
+      if (!newProduct.default_price)
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "New product does not have a default price in Stripe",
+        });
+
+      const prices = await stripeClient.prices.list({
```

Comment:
retrieve() allows to get by id in case we cannot just use the id but need the object